app/models/annotation.rb

```diff
@@ -19,12 +19,15 @@ class Annotation < ApplicationRecord
   validates_format_of :type,
                       with: /\AImageAnnotation|TextAnnotation|PdfAnnotation\z/
 
-  after_create :modify_mark_with_deduction
-  after_destroy :modify_mark_with_deduction
+  after_create :modify_mark_with_deduction, unless: lambda { |a|
```

Comment:
Ah, so here it'll be more concise (and fit on one line) if you do something like `[nil, 0].include? a.annotation_text.deduction`